### ISSUE TITLE
Représentation de la répartition taxonomique dans les rapports d'imports et API Synthèse

### ISSUE THREAD

**Issue by DonovanMaillard** [PnX-SI/gn_module_import#221]
Actuellement, le module métadonnées représente une répartition taxonomique par groupe 2 inpn, et cette représentation sera reprise dans les rapports d'imports à venir (#158 )

On voit que les besoins différent selon les utilisateurs et meme les modules d'après les échanges sur #158 notamment

Ce graphique appelle l'API de la synthèse /api/synthese/taxa_distribution?taxa_rank=group2_inpn&id_dataset=230 et la filtre par group2_INPN et par dataset.

/api/synthese/taxa_distribution est déjà en mesure de répondre à plusieurs rangs taxonomiques (classe, ordre, famille, group2 etc). On envisage donc un paramètre au niveau du module d'imports qui permette de définir le rang à utiliser pour le graphique, et qui pourrait aussi être dupliqué au niveau des métadonnées à terme.

Pour utiliser cette API cependant (qui pour le moment filtre par dataset), on aurait besoin d'y requêter non plus un id_dataset, mais un id_source (puisque 1 import=1 source). @bouttier @TheoLechemia est-ce qu'il vous semble souhaitable d'ajouter ce paramètre id_source aux paramètres de cette API /api/synthese/taxa_distribution ? 

De la même manière, l'API synthèse peut requêter un id_dataset directement via la querystring de son URL. Ca permet d'afficher les données d'un jeu de données en synthèse depuis le module métadonnées. Même question, est-ce qu'il vous semble pertinent d'ajouter le paramètre id_source pour afficher les données d'une source dans la synthèse ? J'y vois un autre intérêt, en allant voir en synthèse toutes les données qui viennent d'un module (CMR, monitoring ou autre) sans passer par les JDD. Ce qui permettrait de récupérer toutes les données de tel module de saisie de protocole plus facilement...

**Comment by Adrien-Pajot** [comment]
> /api/synthese/taxa_distribution est déjà en mesure de répondre à plusieurs rangs taxonomiques (classe, ordre, famille, group2 etc). On envisage donc un paramètre au niveau du module d'imports qui permette de définir le rang à utiliser pour le graphique, et qui pourrait aussi être dupliqué au niveau des métadonnées à terme.

Ce rang ne pourrait-il pas être généré "intelligemment" par le module lui-même. Si rang0 inférieur à 3 alors rang1 si inférieur à 3 alors rang2, etc en choisissant ce que la personne préfère comme limite de rang ?

**Comment by DonovanMaillard** [comment]
Ca me semble assez discutable, on aurait certaines fiches à l'ordre, d'autres à la classe, d'autres au groupe 2, sans pouvoir homogénéiser... J'imagine facilement dans notre instance invertébrés des fiches à la famille pour les lots de données odonates, et des choses à la classe pour des jeux génériques. 

Pour autant, je préfère rester à l'ordre, et afficher que ce jeu comporte 100% d'odonates... 

Par contre, est-ce qu'il ne faut pas prévoir un menu déroulant à coté du Graphique pour que l'utilisateur puisse switcher d'un rang à l'autre, en paramétrant uniquement le rang par défaut... à voir...

**Comment by TheoLechemia** [comment]
@DonovanMaillard, oui  ce n'est pas compliqué d'ajouter un filtre par "id_source" sur la route `synthese/taxa_distribution`
On peut imaginer un bouton pour switcher le rang du graphique et/ou un paramètre pour le rang par défaut de l'instance ?

**Comment by DonovanMaillard** [comment]
Oui, discuté je crois je sur un autre ticket, l'idée serait de parametrer un rang par defaut utilisé sur le pdf +  un bouton pour que l'utilisateur puisse switcher 

Dans l'idée ca veut dire que la version x de limport nécessite la y de geonature et quon doit releaser GN avant l'import a venir
